### PR TITLE
Add markdown and ipynb support

### PR DESCRIPTION
**Pull request by CluckRookie**

### CODE REVIEW

File: uniflow/op/extract/ipynb_op.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        Returns:
+            Sequence[Node]: Nodes after running.
+        """
+        try:
+            import nbformat  # pylint: disable=import-outside-toplevel
+            from nbconvert import (  # pylint: disable=import-outside-toplevel
+                MarkdownExporter,
+            )
+        except ModuleNotFoundError as exc:
+            raise ModuleNotFoundError(
```
Comment (goldmermaid) [inline uniflow/op/extract/ipynb_op.py:28]: Do we need to inform user to install `nbformat` and `nbconvert` during uniflow installation?

File: uniflow/extract/flow/extract_ipynb_flow.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from uniflow.op.extract.ipynb_op import ExtractIpynbOp
+
+
+class ExtractIpynbFlow(Flow):
+    """Extract txt Flow Class."""
+
+    TAG = EXTRACT
+
+    def __init__(self) -> None:
+        """Extract txt Flow Constructor."""
```
Comment (CambioML) [inline uniflow/extract/flow/extract_ipynb_flow.py:17]: nit: should be extract ipynb.

File: uniflow/extract/flow/extract_ipynb_flow.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        super().__init__()
+        self._extract_ipynb_op = ExtractIpynbOp(name="extract_ipynb_op")
+
+    def run(self, nodes: Sequence[Node]) -> Sequence[Node]:
+        """Run Extract txt Flow.
```
Comment (CambioML) [inline uniflow/extract/flow/extract_ipynb_flow.py:22]: nit: same as above.

File: uniflow/extract/flow/extract_pdf_flow.py [hunk 4]
```diff
@@ -27,13 +25,14 @@ def __init__(
             model_config (Dict[str, Any]): Model config.
         """
         super().__init__()
-        self._process_pdf_op = ProcessPDFOp(
-            name="process_pdf_op",
+        self._extract_pdf_op = ExtractPDFOp(
+            name="extract_pdf_op",
             model=PreprocessModel(
-                guided_prompt_template=guided_prompt_template,
+                guided_prompt_template=None,
```
Comment (CambioML) [inline uniflow/extract/flow/extract_pdf_flow.py:31]: You can rebase the latest code. Then, PreprocessModel does not need `guided_prompt_template` argument anymore.

File: uniflow/op/extract/ipynb_op.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import copy
+import re
+from typing import Sequence
+
+from uniflow.node.node import Node
+from uniflow.op.op import Op
+
+
+class ExtractIpynbOp(Op):
+    """Extract txt Op Class."""
```
Comment (CambioML) [inline uniflow/op/extract/ipynb_op.py:11]: nit: it should be ipynb

File: uniflow/op/extract/ipynb_op.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+"""Extract txt op."""
```
Comment (CambioML) [inline uniflow/op/extract/ipynb_op.py:1]: nit: ipynb

File: uniflow/op/extract/ipynb_op.py (hunk 1 above)
Comment (CambioML) [reply uniflow/op/extract/ipynb_op.py:28]: +1

File: uniflow/op/extract/ipynb_op.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                MarkdownExporter,
+            )
+        except ModuleNotFoundError as exc:
+            raise ModuleNotFoundError(
+                "Please install nbformat and nbconvert to load ipynb file. You can use `pip install nbformat nbconvert` to install them."
+            ) from exc
+        output_nodes = []
+        for node in nodes:
+            value_dict = copy.deepcopy(node.value_dict)
+            nb = nbformat.read(value_dict["filename"], as_version=4)
```
Comment (CambioML) [inline uniflow/op/extract/ipynb_op.py:34]: qq: what does `as_version` means here? Should we make `as_version` configurable?

File: uniflow/op/extract/md_op.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
+"""Extract txt op."""
```
Comment (CambioML) [inline uniflow/op/extract/md_op.py:1]: nit: same it should be md.

File: uniflow/op/extract/md_op.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import copy
+import re
+from typing import Sequence
+
+from uniflow.node.node import Node
+from uniflow.op.op import Op
+
+
+class ExtractMarkdownOp(Op):
```
Comment (CambioML) [inline uniflow/op/extract/md_op.py:10]: Do we also need a `MarkdownFlow` flow to use this Op?

**Review by CambioML — COMMENTED** [review]
Can we also add corresponding notebook and also unittest(can just ask copilot to write) to catch potential bugs.

**Pull request closed without merging**